docs/issues.md

```diff
@@ -0,0 +1,147 @@
+## Current issues
+
+### Protocol analisys - early conclusions
+
+Getting STM32 to communicate with
+[VCOM](https://github.com/lpn-plant/ms-tpm-20-ref/tree/master/Samples/Nucleo-TPM/VCOM)
+application leads us to a current yet not resolved problem.
+
+Right now we are able to execute commands on TPM, but an error occurs when the
+host device verifies the response data. Specifically, this line of code causes an
+error:
+```
+*((unsigned int*)&response[sizeof(unsigned short) + sizeof(unsigned int)]) == 0
+```
+
+[VCOM-TPM.cpp:198](https://github.com/lpn-plant/ms-tpm-20-ref/blob/master/Samples/Nucleo-TPM/VCOM/VCOM-TPM/VCOM-TPM.cpp#L198)
+
+For now, it's not clear what the response data of TPM_Startup command should look
+like and what each byte of command represents.
+
+Command and response data looks as follows.
+```
+unsigned char CmdBuf[12] = {
+0x80, 0x01, 0x00, 0x00, 0x00, 0x0c, 0x00, 0x00, 0x01, 0x44, 0x00, 0x00
+};
+
+unsigned char RspBuf[12] = {
+0x80, 0x01, 0x00, 0x00, 0x00, 0x0a, 0x00, 0x00, 0x00, 0x00
+};
+```
+
+This data is output via ITM trace on port 2.
+
+Correctness of data was verified on both sides of the communication.
+
+Startup command data is hardcoded in VCOM application
+[here](https://github.com/lpn-plant/ms-tpm-20-ref/blob/master/Samples/Nucleo-TPM/VCOM/VCOM-TPM/VCOM-TPM.cpp#L192)
+what makes it hard to reason about its origins.
+
+
+Raw data dumped from `CDC_Receive_FS` shows some similarities between VCOM
+sent command and test run of `tpm2_startup -T device:/dev/ttyACM1` application.
+
+VCOM Startup command received
+```
+0x54 0x70 0x6d 0x32 0x3 0x0 0x0 0x0 0x4 0x0 0x0 0x0 0xac 0x44 0x69 0x61 
+0x54 0x70 0x6d 0x32 0x6 0x0 0x0 0x0 0x14 0x0 0x0 0x0 0x0 0x0 0x0 0x0 0xc 0x0 0x0 0x0 0x80 0x1 0x0 0x0 0x0 0xc 0x0 0x0 0x1 0x44 0x0 0x0 
+```
+
+tpm2_startup received command
+```
+0x80 0x1 0x0 0x0 0x0 0xc 0x0 0x0 0x1 0x44 0x0 0x1 
```

Comment:
@krystian-hebel Any idea how to decode these? Please note that (almost) the same pattern from the tpm2_startup appears also in the MS application which was used to test the ms-tpm-20-ref stack

Reply:
Yes, tedious, manual workflow:
- https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-3-Commands-01.38.pdf 9.3.2 has a table with high-level format, also read section  `Notation` in that document - there are important notes about table decorators and other symbols changing the meaning of fields in that table
- definitions of fields in that table can be found in https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf - remember to search for full name, there are names that differ with just one letter